### ISSUE TITLE
rostopic: pub a single message from file

### ISSUE THREAD

**Issue by m-naumann** [ros/ros_comm#1347]
Hello,

when trying to publishing a single message from file, it is published only once, even though a rate is given.

It is certainly desired that, when having multiple messages in the file, they are published one after another at the desired rate. But when providing just one message, it should be repeatedly published.

#### as desired, `argv_publish`:

`rostopic pub -r 1 /test std_msgs/Header '{}'`
-> publishes a header every second

#### misleading, from file via `stdin_publish`
`rostopic pub -r 1 /test std_msgs/Header -f header.yaml`

with header.yaml:

```
  seq: 0
  stamp:
    secs: 0
    nsecs:         0
  frame_id: 'test_frame'
```
-> publishes the message once in latching mode

**Comment by koenlek** [comment]
Same issue here

**Comment by dirk-thomas** [comment]
Addressed in #1635.
